Gravatar Quick Editor as My Profile editor

etoledom opened the pull request:
## Description
This PR changes the previous `MyProfileViewController` with the Gravatar Quick Editor to edit the My Profile fields.

![CleanShot 2025-06-11 at 22 25 03](https://github.com/user-attachments/assets/f5c2520c-b7bc-4ece-86a2-3febb36eb933)

## Testing instructions

- Go to Me tab
- Tap on the avatar view
  - Check that the Quick Editor is presented on Avatar Picker mode
  - Check that selecting a new avatar updates the current avatar on Jetpack side
- Go back
- Tap on My Profile
  - Check that the Quick Editor is presented on About Editor mode (profile fields editor)
- Update all the fields
- Go back
  - Check that the display name has change on the My Profile tab header
  - Check on WP.com that the other fields have been updated acordingly    

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., VoiceOver)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

etoledom committed:
Using Gravatar Quick Editor as My Profile editor

etoledom committed:
Update Gravatar SDK reference

etoledom commented on WordPress/Classes/ViewRelated/Me/My Profile/MyProfileHeaderView.swift:
```diff
@@ -104,9 +104,10 @@ class MyProfileHeaderView: UITableViewHeaderFooterView, NibLoadable {
+        else { return }
+        presenter.presentQuickEditor(on: presentingViewController, scope: .avatarPicker())
     }
 }
```
I believe that this class and `MyProfileViewController` related types are no longer in use.

Should we clean up?

kean replied:
If you could remove `MyProfileViewController` and the related types, that would be ideal.

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean reviewed (commented):
Hey, I tested the editor on iPhone, and it worked well.

There is a little bit of a UX issue due to a large, non-scrollable header that makes the form much smaller than it can be:

<img src="https://github.com/user-attachments/assets/80076bb0-71df-45e5-a28a-74fc5076999a" width="320px">

It's especially visible with accessibility fonts on:

<img src="https://github.com/user-attachments/assets/71f98d7b-e6f6-450f-bc0c-282cc87c988f" width="320px">

The email field also sticks out with large fonts. I'd suggest displaying in inside the header "card" below the name.

I haven't tested the iPad version. I presume the form is shown in a "form" modal?

etoledom committed:
Update Gravatar version

etoledom committed:
Using Avatar & About scopes on Gravatar Quick Editor

etoledom committed:
Removing types not longer used

etoledom committed:
Merge branch 'trunk' into proposal/gravatar-quick-editor-as-profile-editor

etoledom commented:
Thank you @kean for the review!

We implemented a few changes taking into consideration reviews from Android and iOS PRs.
The changes are detailed [here.](https://github.com/Automattic/Gravatar-SDK-iOS/pull/782).

Regarding to your feedback, we have:
- Removed the email from the top of the view, giving more vertical space.
- The Profile card won't increase its size for dynamic type sizes beyond xxxl.
- Removed the Profile card when the keyboard is present and the dynamic type size is Accessibility 3 or higher.

Besides this, I have removed from the project the types which have become obsolete.

We also wanted to mention that we have another alternative grid layout for the Avatar picker. 
The alternatives can be seen [here](https://automattic.github.io/Gravatar-SDK-iOS/gravatarui/documentation/gravatarui/quickeditorarticle#Avatar-Picker-scope). We are currently using the "layout 3": Horizontal scrolling with intrinsic height size.

---
Regarding iPad, yes, it will present as a form modal:

<img src="https://github.com/user-attachments/assets/0013b93b-35f3-4409-bef9-eee798393466" width="40%">

etoledom committed:
Merge branch 'trunk' into proposal/gravatar-quick-editor-as-profile-editor

etoledom commented:
@kean - Ready for another look 👀

kean reviewed (approved):
LGTM

Minor comments:

- I still find it strange that only a small portion of the screen is scrollable while the rest isn’t. Additionally, the scroll indicator seems too far from the right edge. I recommend making the entire screen scrollable, including the header. It would also be helpful to move the Gravatar logo elsewhere and place the "Save" button in the navigation bar. This change would help save vertical space and align with the standard design for these types of forms in the app. If this isn't the direction you want to take, that's okay, but it doesn't follow typical native app conventions.

- The notice is shown too close to the keyboard

<img src="https://github.com/user-attachments/assets/1a765346-3d0a-4376-bbf2-9e4640452fb1" width="320px">


Hey, @crazytonyli , do you want to take a second look at this PR?

crazytonyli commented on Modules/Package.swift:
```diff
@@ -27,7 +27,7 @@ let package = Package(
         .package(url: "https://github.com/Automattic/Automattic-Tracks-iOS", from: "3.5.2"),
         .package(url: "https://github.com/Automattic/AutomatticAbout-swift", from: "1.1.5"),
-        .package(url: "https://github.com/Automattic/Gravatar-SDK-iOS", from: "3.2.0"),
+        .package(url: "https://github.com/Automattic/Gravatar-SDK-iOS", revision: "ce04275c6237131576f424a215197bfaeaefc416"),
```
Just a reminder to update the commit to a tag.

etoledom committed:
Update Gravatar SDK to v3.4.0

etoledom committed:
Merge branch 'trunk' into proposal/gravatar-quick-editor-as-profile-editor

etoledom committed:
Merge branch 'trunk' into proposal/gravatar-quick-editor-as-profile-editor

etoledom committed:
Merge branch 'trunk' into proposal/gravatar-quick-editor-as-profile-editor
